GetStream/stream-chat-swift
# Show system message on channel truncated event

### 🔗 Issue Link
CIS-1503

### 🎯 Goal

* Allow clients to pass system message when truncating the channel
* Parse and display system message attached to the `channel.truncated` event.

### 🛠 Implementation

**Sending the truncate channel request:**
Existing `truncateChannel(:)` function is now synced with the REST endpoint. When system message is specified, the `MessageRequestBody` wrapped in `ChannelTruncatePayloadRequest` is leveraged under the hood to build the body of the HTTP request.

**Displaying system message once the channel is truncated:*
The `message` model is now parsed once the `ChannelTruncatedEvent` is received. The message is then saved in the DB, which in turn updates the UI.

### 🎨 Changes

### 🧪 Testing

1. Create new or enter existing channel
2. Leverage the new debug options to truncate the channel w/o or with the system message
![Uploading Simulator Screen Shot - iPhone 13 Pro - 2022-02-16 at 17.31.30.png…]()
3. Verify the list is emptied once the channel is truncated. If the channel is truncated with the system message, you should spot `Channel truncated` system message at the bottom (as shown on the screenshot).

![Simulator Screen Shot - iPhone 13 Pro - 2022-02-16 at 17 33 12](https://user-images.githubusercontent.com/4924709/154311752-261c5ec8-e609-4f4f-b841-9f6e354bd201.png)

### ☑️ Checklist
